modules/cfe_testcase/src/resource_id_misc_test.c

```diff
@@ -0,0 +1,163 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: resource_id_misc_test.c
+**
+** Purpose:
+**   Functional test of Miscellaneous Resource Id APIs
+**
+**   Demonstration of how to register and use the UT assert functions.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+/* Needed for CFE_RESOURCEID_MAKE_BASE macro */
+#include "cfe_resourceid_basevalue.h"
+
+bool TestReturnFalse(CFE_ResourceId_t CheckId)
+{
+    return false;
+}
+
+bool TestReturnTrue(CFE_ResourceId_t CheckId)
+{
+    return true;
+}
+
+void TestToFromInteger(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_ToInteger, CFE_ResourceId_FromInteger, CFE_ResourceId_Equal");
+    uint32           Id1            = 1;
+    uint32           Id2            = 999;
+    CFE_ResourceId_t ResourceId1    = CFE_ResourceId_FromInteger(Id1);
+    CFE_ResourceId_t ResourceId2    = CFE_ResourceId_FromInteger(Id2);
+    CFE_ResourceId_t ResourceId1Exp = Id1;
+    CFE_ResourceId_t ResourceId2Exp = Id2;
+    UtAssert_UINT32_EQ(CFE_ResourceId_ToInteger(ResourceId1), Id1);
+    UtAssert_UINT32_EQ(CFE_ResourceId_ToInteger(ResourceId2), Id2);
+    UtAssert_UINT32_EQ(ResourceId1, ResourceId1Exp);
+    UtAssert_UINT32_EQ(ResourceId2, ResourceId2Exp);
+}
```

Comment:
I think something may have gotten lost in the flow of this test.  I'd expect to see the following:
1) convert from integer to id and back to integer, confirm pre/post conversion indexes are equal
2) convert from id to integer to id and confirm pre/post conversion ids are equal

Or something similar...

Reply:
done :+1: